### ISSUE TITLE
[5.0.0-alpha.7] "Upload Debug Symbols to Sentry" build step uploads unresolved release metadata on iOS

### ISSUE THREAD

**Issue by AlexanderEggers** [getsentry/sentry-react-native#2581]
**OS:**
- [ ]  Windows
- [x]  MacOS
- [ ]  Linux

**Platform:**
- [x]  iOS
- [ ]  Android

**SDK:**
- [x]  `@sentry/react-native` (>= 1.0.0)
- [ ]  `react-native-sentry` (<= 0.43.2)

**SDK version:** 5.0.0-alpha.7

**`react-native` version:** 0.70.4

**Are you using Expo?**
- [ ]  Yes
- [x]  No

**Are you using sentry.io or on-premise?**
- [x]  sentry.io (SaaS)
- [ ]  on-premise

---
**I have following issue:**

After updating to 5.0.0-alpha.7, I noticed that the xcode build step "Upload Debug Symbols" does not resolve the global `CURRENT_PROJECT_VERSION` properly anymore. Instead of resolving that into the actual current build number, it just takes the property name calling the sentry release `0.1.0+$(CURRENT_PROJECT_VERSION)`. Other release details are broken as well including the package name.

**Steps to reproduce:**
- Update to 5.0.0-alpha.7
- Build ios and ensure that "Upload Debug Symbols" runs successfully
- Go to sentry and you can see that your release contains a lot of unresolved meta data. 

**Actual result:**

Broken release meta details on sentry.io.

**Expected result:**

Proper release meta details on sentry.io.

**Comment by krystofwoldrich** [comment]
Thank you for the information, this might be related to the bump CLI from v1.74.4 to v2.7.0. I'll take a look at it.

**Comment by krystofwoldrich** [comment]
@AlexanderEggers Could you share your `Upload Debug Symbols to Sentry` script?

**Comment by AlexanderEggers** [comment]
@krystofwoldrich this is my current script I am using for the build step: `export SENTRY_PROPERTIES=sentry.properties\n../node_modules/@sentry/cli/bin/sentry-cli upload-dif \"$DWARF_DSYM_FOLDER_PATH\" --force-foreground --log-level debug\n`

That used to work fine for any previous 5.0.0 alpha release - so it surprising that the latest one is breaking that integration. For example using alpha.6, my releases are properly uploaded including the right metadata.

**Comment by krystofwoldrich** [comment]
Thanks for the clarification I thought the name was replaced with a version.

Could you check what version of the CLI you have in `node_modules/@sentry/cli`?

**Comment by AlexanderEggers** [comment]
@krystofwoldrich According to its package.json, version 2.7.0 was installed.

That happened every time regardless how many times I was compiling the app.

**Comment by krystofwoldrich** [comment]
Could you send a small example app for a repro? I can't reproduce it and it seems like we have the same SDK, the same CLI version.

**Comment by AlexanderEggers** [comment]
Unfortunately that will take some time, I will try to take care of that by the end of next week. It is possible that this is very specific to my project, is there anything else I could try to provide which might help you?

**Comment by irothenbaum** [comment]
I'm experiencing the same/similar issue after updating to version 4.7.1, cli @ 2.8.0

For us, the entire release and dist labels are coming through un-interpolated. For example, instead of coming in as `com.example.app@2022.10.27+(1)`, it's coming in as `$(PRODUCT_BUNDLE_IDENTIFIER)@$(MARKINEG_VERISON)+$(CURRENT_PROJECT_VERSION)`

**See screenshot:**
![image](https://user-images.githubusercontent.com/1571538/198663005-535f7106-f166-4f9c-920b-4012820d4b06.png)

**Our Bundle React Native Code and Images script is**
```
export NODE_BINARY=$(which node)
export EXTRA_PACKAGER_ARGS="--sourcemap-output $DERIVED_FILE_DIR/main.jsbundle.map"
export SENTRY_PROPERTIES=sentry.properties

../node_modules/@sentry/cli/bin/sentry-cli react-native xcode --force-foreground
  ../node_modules/react-native/scripts/react-native-xcode.sh
```

**and our Upload Debug Symbols to Sentry script is:**

```
export SENTRY_PROPERTIES=sentry.properties
[[ $SENTRY_INCLUDE_NATIVE_SOURCES == "true" ]] && INCLUDE_SOURCES_FLAG="--include-sources" || INCLUDE_SOURCES_FLAG=""

../node_modules/@sentry/cli/bin/sentry-cli debug-files upload "$INCLUDE_SOURCES_FLAG"
```

**Comment by krystofwoldrich** [comment]
@irothenbaum What command are you using to build your app or are you using Xcode GUI?

**Comment by irothenbaum** [comment]
We're using the Xcode GUI. Product -> Archive

**Comment by Thanaen** [comment]
Hi!
I just want to add that I have the same issue, but I'm building the app with Fastlane.
On the Sentry "Versions" page, my latest version is displayed as "$(PRODUCT_BUNDLE_IDENTIFIER)@x.y.z+XX

**Comment by krystofwoldrich** [comment]
The values should be read from the Xcode environment or `Info.plist` file. When you check `Bundle React Native code and images` and `Upload Debug Symbols to Sentry` build phases, you should see the correct values.

Example:
```bash
...
  export PRODUCT_BUNDLE_IDENTIFIER\=io.sentry.sample
...
```


@Thanaen Could you create a minimal example app where we can reproduce the issue and upload a repo link or archive?

**Comment by marandaneto** [comment]
Any updates here @AlexanderEggers ?

**Comment by krystofwoldrich** [comment]
I've opened PR in `sentry-cli` which fixes this issue.

- https://github.com/getsentry/sentry-cli/pull/1403

**Comment by AlexanderEggers** [comment]
@krystofwoldrich Do you know when this fix will be available in a 4.x or 5.x release?

**Comment by kamilogorek** [comment]
FYI, fix was released in `sentry-cli@2.10.0`.

**Comment by irothenbaum** [comment]
Confirmed this solved it for us. Thank you!
